### PR TITLE
Removed deprecation from Enchantment#isCursed and remove ItemStackRecipeChoice

### PR DESCRIPTION
**Pull request by WillFP**
Very minor change, but Enchantment#isCursed has valid applications, especially considering custom enchantment plugins that may add new curses, that may not contain 'curse' in the enchantment key, which is the other way of checking if an enchantment is a curse (other than hardcoding a list).

ItemStackRecipeChoice is a redundant class since the adding of ExactChoice in RecipeChoice. Given that it was always deprecated, there shouldn't be any usages of it in plugins.

### CODE REVIEW

File: patches/api/0333-Removed-deprecation-from-Enchantment-isCursed.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+diff --git a/src/main/java/org/bukkit/enchantments/Enchantment.java b/src/main/java/org/bukkit/enchantments/Enchantment.java
+index 05f516eb488124a86dcd31a52ffa1f4b847545ac..7fdba0f1eb17345e5f31d687361d88f2810837b0 100644
+--- a/src/main/java/org/bukkit/enchantments/Enchantment.java
++++ b/src/main/java/org/bukkit/enchantments/Enchantment.java
+@@ -272,7 +272,6 @@ public abstract class Enchantment implements Keyed, net.kyori.adventure.translat
+      * only for {@link Enchantment#BINDING_CURSE} and
+      * {@link Enchantment#VANISHING_CURSE}.
+      */
+-    @Deprecated
```
Comment (NoahvdAa) [inline patches/api/0333-Removed-deprecation-from-Enchantment-isCursed.patch]:
```suggestion
-    @Deprecated
+    // @Deprecated // Paper
```
Add a paper comment here.

File: patches/api/0334-Removed-ItemStackRecipeChoice.patch [hunk 2]
```diff
@@ -0,0 +1,63 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Auxilor <william.favierparsons1@gmail.com>
+Date: Sun, 29 Aug 2021 21:16:51 +0100
+Subject: [PATCH] Removed ItemStackRecipeChoice
```
Comment (NoahvdAa) [inline patches/api/0334-Removed-ItemStackRecipeChoice.patch]: Remove the [original patch](https://github.com/PaperMC/Paper/blob/master/patches/api/0153-Add-ItemStackRecipeChoice-Draft-API.patch) instead (and rebuild afterwards).

**Comment by jpenilla** [comment]
This should be two separate PRs

File: patches/api/0334-Removed-ItemStackRecipeChoice.patch (hunk 2 above)
Comment (WillFP) [reply patches/api/0334-Removed-ItemStackRecipeChoice.patch]: on it

File: patches/api/0333-Removed-deprecation-from-Enchantment-isCursed.patch (hunk 1 above)
Comment (WillFP) [reply patches/api/0333-Removed-deprecation-from-Enchantment-isCursed.patch]: on it

**Comment by WillFP** [comment]
i'll be honest fellas i think i've fucked this up

**Pull request closed without merging**